tools/unirestore/scylla_restore.py

```diff
@@ -0,0 +1,996 @@
+
+      It is assumed the file has been received from a DESCRIBE SCHEMA dump, and thus all the
+      keyspace/tables are read from the lines containing "CREATE TABLE ks.table ...".
+      Other formats are not supported.
+
+    Returns:
+      keyspaces (dict): dictionary of keyspace:[list of tables]
+    '''
+    def read_schema_file(filename):
+        try:
+            with open(filename, 'r') as f:
+                schema_data = f.readlines()
+        except Exception as e:
+            logger.error("Failed reading schema file", exc_info=True)
+            sys.exit(1)
+        logger.debug('raw schema file: \n%s', str(schema_data))
+        return schema_data
+    schema = read_schema_file(schema_file)
+    keyspaces = {}
+    for line in schema:
+        if line.upper().startswith('CREATE TABLE'):
+            s = line.split()
+            try:
+                if 'IF EXISTS' in line.upper():
+                    ks = s[4].split('.')[0]
+                    tb = s[4].split('.')[1]
+                else:
+                    ks = s[2].split('.')[0]
+                    tb = s[2].split('.')[1]
+                if ks in keyspaces.keys():
+                    keyspaces[ks].append(tb)
+                else:
+                    keyspaces[ks] = [tb]
+            except IndexError as e:
+                logger.error('Malformed CREATE TABLE line in %s\n%s', line, str(e))
+                sys.exit(1)
+            except Exception as e:
+                logger.error('Error parsing the source schema:\n %s', str(e))
+                sys.exit(1)
+    logger.debug('Read schema from schema dump: %s', str(keyspaces))
+    return keyspaces
+
+def get_destination_dirs(schema, config, ssh_auth):
+    '''
+    Access the system tables in the destination cluster and gather the UUIDs of the table directories.
+    These will be used to determine the destination paths for the restore files.
+
+    This function also makes sure that the restored schema items are present in the destination cluster.
+
+    Args:
+      schema (dict): schema containing the keyspaces and tables for the restored dataset
+      config (dict): configuration values dictionary
+      ssh_auth (dict): dictionary containing ssh authentication parameters
+
+    Returns:
+      dest_dirs (list): list of destination paths for the restored schema
+    '''
+    def accumulate(acc, item):
+        # accumulator function for the reduce()
+        return acc + len(item[1])
+    host_string = ("CQLSH_HOST=%s" % config['cql_host'])
+    cmd = [host_string, 'cqlsh']
+    if config['cql_user'] != 'None':
+        cmd.append("-u " + config['cql_user'])
+    if config['cql_password'] != 'None':
+        cmd.append('-p ' + config['cql_password'])
+    cmd.append('-e "SELECT * FROM system_schema.tables ;"')
+    cmd = ' '.join(cmd)
+    _out = run_remote(cmd, [config['dest_host']], ssh_auth)
+    out = _out[config['dest_host']]['stdout']
+    dest_dirs = []
+    for line in out:
```

Comment:
using scylla-driver or cassandra-driver, can make most of this string parsing code, go away magically...
even a cosmetic change in cqlsh might break it.

Reply:
I wanted to avoid the requirement for anything but ssh access from the restore host to the cluster...

Reply:
Since it's on the table how this is going to be packaged, adding requirements should become easy at certain point, with rpm/deb dependencies or with python setup.py. I would advise doing both

Reply:
@fruch I don't mind doing both, but I might need a bit of help with that. Lets work out the kinks in the code first though